.github/workflows/solidity-foundry-artifacts.yml

```diff
@@ -0,0 +1,479 @@
+        required: true
+
+env:
+  FOUNDRY_PROFILE: ci
+  # Has to match the `make foundry` version.
+  FOUNDRY_VERSION: nightly-de33b6af53005037b463318d2628b5cfcaf39916
+
+jobs:
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.changes.outputs.sol }}
+      product_changes: ${{ steps.changes-transform.outputs.product_changes }}
+      files: ${{ steps.changes-transform.outputs.product_files }}
+      changeset_changes: ${{ steps.changes-dorny.outputs.changeset }}
+      changeset_files: ${{ steps.changes-dorny.outputs.changeset_files }}
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Find modified contracts
+        uses: AurorNZ/paths-filter@3b1f3abc3371cca888d8eb03dfa70bc8a9867629 # v4.0.0
+        id: changes
+        with:
+          list-files: 'csv'
+          base: ${{ inputs.base_ref }}
+          filters: |
+            ignored: &ignored
+              - '!contracts/src/v0.8/**/test/**'
+              - '!contracts/src/v0.8/**/tests/**'
+              - '!contracts/src/v0.8/**/mock/**'
+              - '!contracts/src/v0.8/**/mocks/**'
+              - '!contracts/src/v0.8/**/*.t.sol'
+              - '!contracts/src/v0.8/*.t.sol'
+              - '!contracts/src/v0.8/**/testhelpers/**'
+              - '!contracts/src/v0.8/testhelpers/**'
+              - '!contracts/src/v0.8/vendor/**'
+            other_shared:
+              - 'contracts/src/v0.8/interfaces/**/*.sol'
+              - 'contracts/src/v0.8/*.sol'
+              - *ignored
+            sol:
+              - 'contracts/src/v0.8/**/*.sol'
+              - *ignored   
+            product: &product
+              - 'contracts/src/v0.8/${{ inputs.product }}/**/*.sol'      
+              - *ignored
+
+      - name: Find modified changesets
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes-dorny
+        with:
+          list-files: 'shell'
+          base: ${{ inputs.base_ref }}
+          filters: |
+            changeset:
+              - modified|added: 'contracts/.changeset/!(README)*.md'
+
+      # TODO remove when https://github.com/AurorNZ/paths-filter/issues/12 is fixed and we can use modifiers with it
+      - name: Get reference to compare to
+        id: get-base-ref
+        shell: bash
+        run: |
+          type=$(git cat-file -t "${{ inputs.base_ref }}")
+          if echo "$type" | grep -q commit; then
+              # we do not know to which branch the base ref belongs, so we fetch all branches
+              git fetch --all --prune
+              ref_to_use="${{ inputs.base_ref }}"
+          elif  echo "$type" | grep -q tag; then
+              git fetch origin ${{ inputs.base_ref }}
+              ref_to_use="origin/${{ inputs.base_ref }}" 
+          else
+              # must be branch
+              echo "::debug:: Base reference type: $type"
```

Comment:
I would add this debug statement right below line 86.

Reply:
not needed any more due to `predicate-quantifier: every`